### ISSUE TITLE
Labels of the full-connectome matric with the aparc-a2009s parcellation 

### ISSUE THREAD

**Issue by jbdavion** [MICA-MNI/micapipe#130]
Those issues could be related : #124 #106 #119

Hello

We are performing structural connectome (SC) analysis with micapipe. The full-SC shape with the aparc-a2009s parcellation has a shape 198 x 198 (this is also precised here : https://micapipe.readthedocs.io/en/latest/pages/04.fetch/index.html?highlight=load_annot), which includes cerebellar, subcortical and cortical nodes (precised here : https://micapipe.readthedocs.io/en/latest/pages/04.matrices/index.html?highlight=full%20connectome#structural-connectome). 

Is you look at the connectome aspect and look-up tables from micapipe, we can infer that the connectome is composed of (in order) : 
- 48 subcortical and cerebellar ROI (infered from lut_subcortical-cerebllum_mics.csv)
- 150 cortical regions (infered from lut_subcortical-cerebllum_mics.csv), 75 for the left hemisphere (starting from ctx_lh_G_and_S_frontomargin to ctx_lh_S_temporal_transverse) and 75 for the righ hemisphere (starting from ctx_rh_G_and_S_frontomargin to ctx_rh_S_temporal_transverse) 
Indeed 48 + 150 = 198, so this is consistent.

in this webpage (https://micapipe.readthedocs.io/en/latest/pages/04.fetch/index.html?highlight=load_annot), a function load_sc is defined to load the SC in order to plot it on a figure, without the medial wall. We are a bit confused by this part of the function : 

```
  # slice the matrix remove subcortical nodes and cerebellum
    SC = mtx_sc[49:, 49:]
    SC = np.delete(np.delete(SC, Ndim, axis=0), Ndim, axis=1)
```

The line `SC = mtx_sc[49:, 49:]` slices the 48 first items which are the subcortical and cerebellar nodes (index from 0 to 47), but please why do you slice the 49th (index 48, which should be ctx_lh_G_and_S_frontomargin) ? Identically, the line `SC = np.delete(np.delete(SC, Ndim, axis=0), Ndim, axis=1) ` slices the 76th item (Ndim = 75 for the aparc-a2009s parcellation), which seems to be ctx_rh_G_and_S_frontomargin. 

Is the 75 cortical areas well the one included in the lut_subcortical-cerebllum_mics.csv please ? More generally, I think look-up tables for the full connectome in all parcellations would be of great help for micapipe users.

Thank you for your help, and for this great tool.

**Comment by wpf1027** [comment]
In the output of the micapipe using the Schaefer-400 parcellation, the original matrix has dimensions 450×450. In this matrix, the first 48 nodes correspond to subcortical structures, while the remaining 402 nodes represent cortical regions. Among these 402 cortical regions, the first and 202nd nodes correspond to the medial wall (i.e., invalid regions). Therefore:

Slice the connectome to use only cortical nodes
SC = mtx_sc[49:, 49:] — This step removes the first 48 subcortical nodes as well as the 49th node (which is actually the first cortical region in the set of 402 nodes), corresponding to the first medial wall.
SC = np.delete(np.delete(SC, 200, axis=0), 200, axis=1) — This step removes the 201st node (which is the 202nd cortical region) from the new matrix obtained in the previous step, corresponding to the second medial wall.
https://micapipe.readthedocs.io/en/latest/pages/04.gradients/index.html
https://micapipe.readthedocs.io/en/latest/pages/04.fetch/index.html?highlight=load_annot
https://github.com/MICA-MNI/micapipe/blob/master/parcellations/lut/lut_schaefer-400_mics.csv

**Comment by RomainViardLille** [comment]
there is no consensus @jbdavion said taht the 49th item is ctx_lh_G_and_S_frontomargin and @wpf1027 said that the 49th item is the first medial wall....who is right ?

**Comment by i-Zaak** [comment]
As far as I can see, the node ordering is created in the  [functions/connectome_slicer.py](https://github.com/MICA-MNI/micapipe/blob/3e5f0c7a68f24469565e79a5731af69c179c5ba8/functions/connectome_slicer.py#L53) as:

```python
lut1 = pd.read_csv(args.lut1)
lut2 = pd.read_csv(args.lut2)
indx = sorted(list(set(lut1['mics']).union(set(lut2['mics']))))
```
where `lut1` is the cerebellum and subcortical areas nad `lut2` is the atlas.

The [parcellations/lut/lut_subcortical-cerebellum_mics.csv](https://github.com/MICA-MNI/micapipe/blob/3e5f0c7a68f24469565e79a5731af69c179c5ba8/parcellations/lut/lut_subcortical-cerebellum_mics.csv) has 48 entries.

As mentioned above, the [parcellations/lut/lut_schaefer-200_mics.csv](https://github.com/MICA-MNI/micapipe/blob/3e5f0c7a68f24469565e79a5731af69c179c5ba8/parcellations/lut/lut_schaefer-200_mics.csv) contains `medial_wall` on positions 0 and 101.

Hence, to get the cortical structural connectivity from full `SC` matrix, one can do something like:

```python
n_sub = 48
n_hemi = 100
s = np.r_[1:n_hemi+1, n_hemi+2:(2*n_hemi+2)]+n_sub
SC = SC[s,:][:,s]
```

**Comment by i-Zaak** [comment]
But it would be helpful if the node labels would be part of the output metadata (in the gifti file or as json sidecar)...

**Comment by RomainViardLille** [comment]
thank for this clarification, so If I can summarise : 
- There is no issue if we use **schaefer_200** parcellation to built our functionnal connectome (FC)
- But if we use **lut/lut_aparc-a2009s_mics.csv** to built FC, the matrix is spurious because the medial wall is not on position 0 and 101 but something like 41 and 116...

**Comment by i-Zaak** [comment]
Yes, there is no issue, per se - just be careful. For example the [parcellations/lut/lut_aparc-a2009s_mics.csv](https://github.com/MICA-MNI/micapipe/blob/3e5f0c7a68f24469565e79a5731af69c179c5ba8/parcellations/lut/lut_aparc-a2009s_mics.csv) you mention, has the medial wall on position 42 and 117. 

And a short disclaimer : I'm just a user, a word on best practices or inteded use from the authors of the pipeline might be useful here @rcruces @alexander-ngo
